### PR TITLE
Local users

### PR DESCRIPTION
**Pull request by yoshiemuranaka**
- Delete user working
- Add/edit user working
- No validations or checks

### CODE REVIEW

File: src/views/AccessControl/LocalUserManagement/LocalUserManagement.vue [hunk 1]
```diff
@@ -1,83 +1,91 @@
...
+              variant="link"
               :disabled="!data.value.edit"
-              v-b-modal.modal-user-settings
+              @click="initModalUser(data.item)"
             >
-              <Edit20 />
+              <icon-edit />
             </b-button>
             <b-button
+              variant="link"
```
Comment (derick-montague) [inline src/views/AccessControl/LocalUserManagement/LocalUserManagement.vue:35]: Can we add aria-labels or visually hidden text for the edit and delete buttons in the table?

File: src/views/AccessControl/LocalUserManagement/ModalUser.vue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  data() {
+    return {
+      privilegeTypes: ["Administrator", "Operator", "ReadOnly", "Callback"]
+    };
+  },
+  computed: {
+    newUser() {
+      return this.user ? false : true;
+    },
+    form() {
```
Comment (derick-montague) [inline src/views/AccessControl/LocalUserManagement/ModalUser.vue:60]: Is there a more intuitive name for this function to better describe what it does?

**Review by derick-montague — COMMENTED** [review]
I noticed some errors when trying to edit user name.  Do we want to use the pattern of closing and showing the toast and updating the table on a successful response? I'll add a story to adding toast messages.

File: src/views/AccessControl/LocalUserManagement/TableRoles.vue [hunk 3]
```diff
@@ -35,80 +35,80 @@ export default {
       items: [
         {
           description: "Configure components managed by this service",
-          admin: true,
+          administrator: true,
           operator: false,
-          user: false,
+          readonly: false,
           callback: false
```
Comment (gtmills) [inline src/views/AccessControl/LocalUserManagement/TableRoles.vue]: This got renamed to NoAccess. 
Need a master build from the last month to see this. 

curl -k https://${bmc}/redfish/v1/AccountService/Roles
{
  "@odata.context": "/redfish/v1/$metadata#RoleCollection.RoleCollection",
  "@odata.id": "/redfish/v1/AccountService/Roles",
  "@odata.type": "#RoleCollection.RoleCollection",
  "Description": "BMC User Roles",
  "Members": [
    {
      "@odata.id": "/redfish/v1/AccountService/Roles/Administrator"
    },
    {
      "@odata.id": "/redfish/v1/AccountService/Roles/Operator"
    },
    {
      "@odata.id": "/redfish/v1/AccountService/Roles/ReadOnly"
    },
    {
      "@odata.id": "/redfish/v1/AccountService/Roles/NoAccess"
    }

File: src/views/AccessControl/LocalUserManagement/ModalUser.vue (hunk 2 above)
Comment (yoshiemuranaka) [reply src/views/AccessControl/LocalUserManagement/ModalUser.vue:60]: What do you have in mind..formData()?

File: src/store/modules/AccessControl/LocalUserMangementStore.js [hunk 4]
```diff
@@ -19,19 +19,42 @@ const LocalUserManagementStore = {
...
+        .catch(error => console.log(error));
+    },
+    updateUser(_, { originalUsername, username, password, privilege, status }) {
+      const data = {};
+      if (username) data.UserName = username;
+      if (password) data.Password = password;
+      if (privilege) data.RoleId = privilege;
+      if (status !== undefined) data.Enabled = status;
+      api
+        .patch(`/redfish/v1/AccountService/Accounts/${originalUsername}`, data)
```
Comment (gtmills) [inline src/store/modules/AccessControl/LocalUserMangementStore.js:52]: Awesome stuff! 

This function could also be used to change the password on expired password on login? 
Any thoughts on how we would share it in this case?

**Review by gtmills — COMMENTED** [review]
Nice work! Long term will need a separate view for Operator / readonly roles. These roles can not "Configure users and their accounts", they can however change thier own password.

File: src/views/AccessControl/LocalUserManagement/TableRoles.vue [hunk 5]
```diff
@@ -35,80 +35,80 @@ export default {
...
           description: "Update password for current user account",
-          admin: true,
+          administrator: true,
           operator: true,
-          user: true,
+          readonly: true,
           callback: false
         },
         {
           description: "Update password for current user account",
```
Comment (gtmills) [inline src/views/AccessControl/LocalUserManagement/TableRoles.vue]: isn't this a repeat of the one above ?

File: src/views/AccessControl/LocalUserManagement/LocalUserManagement.vue (hunk 1 above)
Comment (yoshiemuranaka) [reply src/views/AccessControl/LocalUserManagement/LocalUserManagement.vue:35]: Done

File: src/views/AccessControl/LocalUserManagement/TableRoles.vue (hunk 3 above)
Comment (yoshiemuranaka) [reply src/views/AccessControl/LocalUserManagement/TableRoles.vue]: Done

File: src/views/AccessControl/LocalUserManagement/TableRoles.vue (hunk 5 above)
Comment (yoshiemuranaka) [reply src/views/AccessControl/LocalUserManagement/TableRoles.vue]: Done, thanks for catching

File: src/store/modules/AccessControl/LocalUserMangementStore.js (hunk 4 above)
Comment (yoshiemuranaka) [reply src/store/modules/AccessControl/LocalUserMangementStore.js:52]: Would be the same as what we have in Angular right? Would just need to reference this store module in the login component when password reset is required

Comment (gtmills) [reply src/store/modules/AccessControl/LocalUserMangementStore.js:52]: Sounds good.

File: src/views/AccessControl/LocalUserManagement/ModalUser.vue (hunk 2 above)
Comment (derick-montague) [reply src/views/AccessControl/LocalUserManagement/ModalUser.vue:60]: I think this is fine. :) Great work!

**Review by derick-montague — APPROVED** [review]

**Comment by derick-montague** [comment]
Can you add a story for `No validations or checks`?

**Comment by derick-montague** [comment]
Integrated to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28637

**Comment by derick-montague** [comment]
Migrated to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28731/1

**Pull request merged**